pkg/event/target/queuestore.go

```
@@ -0,0 +1,219 @@
+	}
+
+	return nil
+}
+
+// write - writes event to the directory.
+func (store *QueueStore) write(directory string, key string, e event.Event) error {
+
+	tp := filepath.Join(tmpPath(store.directory), key)
+
+	// Creates the temporary file.
+	temp, cerr := os.Create(tp)
+	if cerr != nil {
+		return cerr
+	}
+
+	// Marshalls the event.
+	eventData, err := json.Marshal(e)
+	if err != nil {
+		return err
+	}
+
+	// Writes to the temporary file.
+	if _, err := temp.Write(eventData); err != nil {
+		return err
+	}
+
+	p := filepath.Join(store.directory, key+eventExt)
+	// Rename the temp file to the destination file.
+	if err := os.Rename(tp, p); err != nil {
+		return err
+	}
+
+	// Increment the event count.
+	store.eC += 1
+
+	return nil
+}
+
+// Put - puts a event to the store.
+func (store *QueueStore) Put(e event.Event) error {
+	store.Lock()
+	defer store.Unlock()
+	if store.eC >= store.limit {
+		return ErrLimitExceeded
+	}
+	key := strconv.Itoa(time.Now().Nanosecond())
+	return store.write(store.directory, key, e)
+}
+
+// Get - gets a event from the store.
+func (store *QueueStore) Get(key string) (event.Event, error) {
+	store.RLock()
+	defer store.RUnlock()
+
+	var event event.Event
+
+	filepath := filepath.Join(store.directory, key)
+
+	if _, err := os.Stat(filepath); err != nil {
+		return event, err
+	}
+
+	// Open the file.
+	eFile, oerr := os.Open(filepath)
+	if oerr != nil {
+		return event, oerr
+	}
+
+	eventData := make([]byte, 1024*2)
+	// Read the file.
+	n, rerr := eFile.Read(eventData)
+	if rerr != nil {
+		return event, rerr
+	}
+
+	uerr := json.Unmarshal(eventData[:n], &event)
+	if uerr != nil {
+		return event, uerr
+	}
+
+	return event, nil
+}
+
+// Del - Deletes an entry from the store.
+func (store *QueueStore) Del(key string) {
+	store.Lock()
+	defer store.Unlock()
+	store.del(key)
+}
+
+// lockless call
+func (store *QueueStore) del(key string) {
+	p := filepath.Join(store.directory, key)
+
+	if _, err := os.Stat(p); err != nil {
```

No need to os.Stat() before removing, just remove, if the file doesn't exist and removed already just ignore the error. So it is quite simply can be one line.

```
os.Remove(filepath.Join(store.directory, key+eventExt))
```